mne_features/univariate.py

```diff
@@ -251,114 +252,94 @@ def compute_hurst_exponent(data):
+    data : ndarray, shape (n_channels, n_times)
+
+    emb : int (default: 2)
+        Embedding dimension.
+
+    metric : str (default: chebyshev)
+        Name of the metric function used with KDTree. The list of available
+        metric functions is given by: `KDTree.valid_metrics`.
 
     Returns
     -------
-    output : ndarray, shape (n_channels)
+    output : ndarray, shape (n_channels,)
 
     References
     ----------
-    .. [1] Teixeira, C. A. et al. (2011). EPILAB: A software package for
-           studies on the prediction of epileptic seizures. Journal of
-           Neuroscience Methods, 200(2), 257-271.
+    .. [1] Borowska, M. (2015). Entropy-based algorithms in the analysis of
+           biomedical signals. Studies in Logic, Grammar and Rhetoric,
+           43(1), 21-32.
     """
+    _all_metrics = KDTree.valid_metrics
+    if metric not in _all_metrics:
+        raise ValueError('The given metric (%s) is not valid. The valid '
+                         'metric names are: %s' % (metric, _all_metrics))
     n_channels, n_times = data.shape
-    sampen = np.empty((n_channels,), dtype=data.dtype)
-    for t in range(n_channels):
-        m = 0
-        s = 0
-        for j in range(n_times):
-            m += data[t, j]
-            s += data[t, j] ** 2
-        m /= n_times
-        s /= n_times
-        s = sqrt(s)
-        x_new = np.zeros(n_times)
-        for j in range(n_times):
-            x_new[j] = (data[t, j] - m) / s
-        mm = 3
-        r = 0.2
-        lastrun = np.zeros((n_times,))
-        run = np.zeros((n_times,))
-        a = np.zeros((mm,))
-        b = np.zeros((mm,))
-        for i in range(n_times - 1):
-            nj = n_times - i - 1
-            y1 = x_new[i]
-            for jj in range(nj):
-                j = jj + i + 1
-                if abs(x_new[j] - y1) < r:
-                    run[jj] = lastrun[jj] + 1
-                    m1 = int(min(mm, run[jj]))
-                    for k in range(m1):
-                        a[k] += 1
-                        if j < (n_times - 1):
-                            b[k] += 1
-                else:
-                    run[jj] = 0
-            for jj in range(nj):
-                lastrun[jj] = run[jj]
-        sampen[t] = -log(a[-1] / b[mm - 2])
-    return sampen
+    phi = np.empty((n_channels, 2))
+    for j in range(n_channels):
+        r = 0.2 * np.std(data[j, :], axis=-1, ddof=1)
+        # compute phi(emb, r)
+        emb_data1 = embed(data[j, None], emb, 1)[0, :, :]
+        count = KDTree(emb_data1, metric=metric).query_radius(
+            emb_data1, r, count_only=True).astype(np.float64)
+        phi[j, 0] = np.mean(np.log(count / emb_data1.shape[0]))
+        # compute phi(emb + 1, r)
+        emb_data2 = embed(data[j, None], emb + 1, 1)[0, :, :]
+        count = KDTree(emb_data2, metric=metric).query_radius(
+            emb_data2, r, count_only=True).astype(np.float64)
+        phi[j, 1] = np.mean(np.log(count / emb_data2.shape[0]))
```

Comment:
those 2 functions seems almost the same. avoid a copy paste by using a private auxiliary function